Added verify_flash option

newinnovations opened the pull request:
I have added the option to verify a portion of the flash against the image on disk.

projectgus commented:
Thanks for this, very useful feature!

A couple of minor comments:
* Could you please squash these commits in your branch and re-push (force push) as a single commit?
* Is it possibe please to print out the address of the first different byte, to help people diagnose problems as being invalid flash as opposed to just bad serial comms or something else?

newinnovations commented:
I have squashed the commits and added some more information when verify fails: number of differences and first failed address. Furthermore there is now an option to dump all failed locations: --diff=yes. Default is no.

projectgus commented:
Thanks for that! Sorry for the delay in updating.

Have merged, plus some additions and refactoring to add a couple of other options I had mind.
